diff --git a/app-config.yaml b/app-config.yaml
index c595b021..6920c3c5 100644
--- a/app-config.yaml
+++ b/app-config.yaml
@@ -18,7 +18,10 @@ backend:
     # Uncomment the following host directive to bind to specific interfaces
     # host: 127.0.0.1
   csp:
-    connect-src: ["'self'", 'http:', 'https:']
+    frame-ancestors: ['http://localhost:3000', 'http://localhost:7007']
+    script-src: ["'self'", "'unsafe-inline'", "'unsafe-eval'"]
+    script-src-elem: ["'self'", "'unsafe-inline'", "'unsafe-eval'"]
+    connect-src: ["'self'", 'http:', 'https:', 'data:']
     # Content-Security-Policy directives follow the Helmet format: https://helmetjs.github.io/#reference
     # Default Helmet Content-Security-Policy values can be removed by setting the key to false
   cors:
@@ -80,3 +83,13 @@ catalog:
 
 dynamicPlugins:
   frontend: {}
+orchestrator:
+  sonataFlowService:
+    baseUrl: http://localhost
+    port: 8899
+    autoStart: true
+    workflowsSource:
+      gitRepositoryUrl: https://github.com/parodos-dev/backstage-orchestrator-workflows
+      localPath: /tmp/orchestrator/repository
+  dataIndexService:
+    url: http://localhost:8899
diff --git a/packages/app/package.json b/packages/app/package.json
index 8685f3a5..76edac42 100644
--- a/packages/app/package.json
+++ b/packages/app/package.json
@@ -57,7 +57,8 @@
     "react-dom": "18.0.0",
     "react-router": "6.23.0",
     "react-router-dom": "6.23.0",
-    "tss-react": "4.9.10"
+    "tss-react": "4.9.10",
+    "@janus-idp/backstage-plugin-orchestrator": "*"
   },
   "devDependencies": {
     "@backstage/cli": "0.27.1",
diff --git a/packages/app/src/App.tsx b/packages/app/src/App.tsx
index d601a5bd..4c4ed635 100644
--- a/packages/app/src/App.tsx
+++ b/packages/app/src/App.tsx
@@ -34,6 +34,8 @@ import { UserSettingsPage } from '@backstage/plugin-user-settings';
 
 import { getThemes } from '@redhat-developer/red-hat-developer-hub-theme';
 
+import { OrchestratorPage } from '@janus-idp/backstage-plugin-orchestrator';
+
 import { apis } from './apis';
 import { entityPage } from './components/catalog/EntityPage';
 import { Root } from './components/Root';
@@ -85,6 +87,8 @@ const routes = (
     </Route>
     <Route path="/create" element={<ScaffolderPage />} />
     <Route path="/api-docs" element={<ApiExplorerPage />} />
+    <Route path="/orchestrator" element={<OrchestratorPage />} />
+
     <Route
       path="/catalog-import"
       element={
diff --git a/packages/app/src/components/Root/Root.tsx b/packages/app/src/components/Root/Root.tsx
index fa83e4d3..143befd2 100644
--- a/packages/app/src/components/Root/Root.tsx
+++ b/packages/app/src/components/Root/Root.tsx
@@ -28,6 +28,8 @@ import { makeStyles } from 'tss-react/mui';
 
 import LogoFull from './LogoFull';
 import LogoIcon from './LogoIcon';
+import { OrchestratorIcon } from '@janus-idp/backstage-plugin-orchestrator';
+import { IconComponent } from '@backstage/core-plugin-api';
 
 const useSidebarLogoStyles = makeStyles()({
   sidebarLogo: {
@@ -82,6 +84,11 @@ export const Root = ({
         <SidebarDivider />
         <SidebarScrollWrapper>
           {/* Items in this group will be scrollable if they run out of space */}
+          <SidebarItem
+           icon={OrchestratorIcon as IconComponent}
+           to="orchestrator"
+           text="Orchestrator"
+          />
         </SidebarScrollWrapper>
       </SidebarGroup>
       <SidebarSpace />
diff --git a/packages/backend/package.json b/packages/backend/package.json
index cdfb2aa2..37a9a5be 100644
--- a/packages/backend/package.json
+++ b/packages/backend/package.json
@@ -42,7 +42,8 @@
     "@janus-idp/backstage-plugin-rbac-backend": "*",
     "@manypkg/get-packages": "1.1.3",
     "app": "*",
-    "winston": "3.11.0"
+    "winston": "3.11.0",
+    "@janus-idp/backstage-plugin-orchestrator-backend": "*"
   },
   "devDependencies": {
     "@backstage/cli": "0.27.1",
diff --git a/packages/backend/src/index.ts b/packages/backend/src/index.ts
index fa1ad154..7a69eb47 100644
--- a/packages/backend/src/index.ts
+++ b/packages/backend/src/index.ts
@@ -45,6 +45,7 @@ backend.add(
 
 // permission plugin
 backend.add(import('@janus-idp/backstage-plugin-rbac-backend'));
+backend.add(import('@janus-idp/backstage-plugin-orchestrator-backend'));
 
 // search plugin
 backend.add(import('@backstage/plugin-search-backend/alpha'));
